File: springboot-app/src/main/java/com/lumos/lumosspring/stock/controller/dto/MaterialResponse.java
package com.lumos.lumosspring.stock.controller.dto;

import com.fasterxml.jackson.annotation.JsonInclude;
import com.lumos.lumosspring.stock.entities.Material;
import com.lumos.lumosspring.stock.entities.MaterialStock;

@JsonInclude(JsonInclude.Include.NON_NULL)  // Isso vai garantir que valores nulos não sejam serializados
public record MaterialResponse(long idMaterial, String materialName, String materialBrand, String materialPower, String materialAmps, String materialLength,
                               String buyUnit, String requestUnit, double stockQt, Boolean inactive, String materialType, String materialGroup,
                               String deposit, String company) {
    public MaterialResponse(MaterialStock material) {
        this(
                material.getMaterialIdStock(),
                material.getMaterial().getMaterialName(),
                material.getMaterial().getMaterialBrand(),
                material.getMaterial().getMaterialPower(),
                material.getMaterial().getMaterialAmps(),
                material.getMaterial().getMaterialLength(),
                material.getMaterial().getBuyUnit(),
                material.getMaterial().getRequestUnit(),
                material.getMaterial().getStockQuantity(),
                material.getMaterial().isInactive(),
                material.getMaterial().getMaterialType() != null ? material.getMaterial().getMaterialType().getTypeName() : null,
                material.getMaterial().getMaterialType() != null ? material.getMaterial().getMaterialType().getGroup().getGroupName(): null,
                material.getDeposit() != null ? material.getDeposit().getDepositName() : null,
                material.getCompany() != null ? material.getCompany().getCompanyName() : null
        );
    }
}

File: springboot-app/src/main/java/com/lumos/lumosspring/stock/materialreference/dto/MaterialResponse.kt
package com.lumos.lumosspring.stock.materialreference.dto

import java.math.BigDecimal

data class MaterialResponse(
    val idMaterial: Long,
    val materialName: String,
    val materialBrand: String?,
    val materialPower: String?,
    val materialAmps: String?,
    val materialLength: String?,
    val buyUnit: String,
    val requestUnit: String,
    val stockQt: BigDecimal,
    val inactive: Boolean,
    val materialType: String,
    val materialGroup: String,
    val deposit: String,
    val company: String
)
